src/MouseAdapter.js

```diff
@@ -201,6 +202,77 @@ define(function(require) {
     }
 
     //---------------------------------------------------------
+    // The Scroll-detection mechanism depends on collecting and storing a small buffer 
+    // of scroll history in order to analyze and interpret it for patterns.
+    //---------------------------------------------------------
+
+    /**
+     * A constant which represents the maximum amount of data buffered.  That is, when 
+     * this value is N, the N most recent data points are preserved.  Beyond that, data 
+     * is overwritten.  This should only be large enough to hold the desired amount of 
+     * data and not any larger to preserve memory.
+     *
+     * This number is 10 because one detection method (_detectUserScroll.detectDeltaIncrease)
+     * uses the last 10 data points to determine whether a scroll was performed.
+     * @type {number}
+     */
+    var DELTA_ARRAY_SIZE = 10;
+    
+    /**
+     * A two dimensional array of size [2][DELTA_ARRAY_SIZE] which holds data captured.  
+     * It holds separate data for the X- and Y-axis, hence the 2.  Constants have been 
+     * created for accessing the array for clarity.
+     * @type {Array.<Array.<number>>}
+     */
+    var deltaArray = [[],[]];
+    
+    /**
+     * An index which always refers to the first (oldest) data point
+     * in the array.  It also represents the point that will be overwritten by new data.
+     * @type {number}
+     */
+    var deltaArrayIndex = 0;
+    
+    /**
+     * A constant provided for accessing the X and Y arrays of data.
+     * @type {number}
+     */
+    var DELTA_INDEX = {
+        x: 0,
+        y: 1
+    };
```

Comment:
I think it'd be easier to debug and less error prone to just have two parallel arrays, one for xs and one for ys. The disadvantage is that you don't see the x and y value together, but I think more often you'd be interested in the trend of x values or y values, not the precise pairing of values.